include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -139,10 +139,29 @@ MultigridPreconditionerBase<dim, Number>::initialize_levels(unsigned int const d
 
   std::vector<unsigned int> h_levels;
 
+
   // setup h-levels
-  unsigned int const n_h_levels = get_number_of_h_levels();
-  for(unsigned int h = 0; h < n_h_levels; h++)
-    h_levels.push_back(h);
+  if(data.involves_h_transfer())
+  {
+    if(multigrid_variant == MultigridVariant::LocalSmoothing)
+    {
+      for(unsigned int h = 0; h < grid->triangulation->n_global_levels(); h++)
+        h_levels.push_back(h);
+    }
+    else if(multigrid_variant == MultigridVariant::GlobalCoarsening)
+    {
+      for(unsigned int h = 0; h < grid->coarse_triangulations.size() + 1; h++)
```

Comment:
why `+1`?

Reply:
`coarse_triangulations` does not contain the fine level triangulation. This was changed in a previous PR, see https://github.com/exadg/exadg/pull/451. Do you agree then?

Reply:
I see :+1: